org.lflang/src/org/lflang/FileConfig.java

```diff
@@ -97,6 +99,8 @@ public abstract class FileConfig {
      */
     public final boolean useHierarchicalBin;
 
+    public final CppFileConfig.CppFiles cpp;
```

Comment:
I don't think that the file config should contain target specific extensions. Why does this change from the previous design? I think the target specific file config classes should extend the base class. We can create the correct one when creating the Context, but only store a reference to `FileConfig`. Then we can still cast to the target specific version in the generator.

If that does not work seems like a bad design, we could alternatively make `FileConfig` and `CppFileConfig` completely separated classes. The C++ generator could then internally create an instance of `CppFileConfig`.

Reply:
Yes, it will either have to be casting or creating a completely separate class. Do you have a preference for either?

Reply:
Fixed in 1cc48459e.